pkg/tfgen/generate.go

```diff
@@ -201,6 +201,125 @@ type moduleMember interface {
 	Doc() string
 }
 
+type typeKind int
+
+const (
+	kindInvalid = iota
+	kindBool
+	kindInt
+	kindFloat
+	kindString
+	kindList
+	kindMap
+	kindSet
+	kindObject
+)
+
+// propertyType represents a non-resource, non-datasource type. Property types may be simple
+type propertyType struct {
+	name       string
+	doc        string
+	kind       typeKind
+	element    *propertyType
+	properties []*variable
+
+	typ        tokens.Type
+	nestedType tokens.Type
+	altTypes   []tokens.Type
+	asset      *tfbridge.AssetTranslation
+}
+
+func makePropertyType(sch *schema.Schema, info *tfbridge.SchemaInfo, out bool, parsedDocs parsedDoc) *propertyType {
```

Comment:
This centralizes the handling of `MaxItemsOne`, `TypeMap` with a `*schema.Resource` element, and doc attachment. These changes provide the most important simplifications in the downstream code.